packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/firewall_match.yml

```
@@ -5,138 +5,167 @@ processors:
-      target_field: "rule.name"
-      type: string
+      tag: rename_image_file_name
+  - rename:
+      field: crowdstrike.event.RuleId
+      target_field: rule.id
       ignore_missing: true
-      ignore_failure: true
-  - convert:
-      field: "crowdstrike.event.RuleGroupName"
-      target_field: "rule.ruleset"
-      type: string
+      tag: rename_rule_id
+  - rename:
+      field: crowdstrike.event.RuleName
+      target_field: rule.name
       ignore_missing: true
-      ignore_failure: true
-  - convert:
-      field: "crowdstrike.event.RuleDescription"
-      target_field: "rule.description"
-      type: string
+      tag: rename_rule_name
+  - rename:
+      field: crowdstrike.event.RuleGroupName
+      target_field: rule.ruleset
       ignore_missing: true
-      ignore_failure: true
-  - convert:
-      field: "crowdstrike.event.RuleFamilyID"
-      target_field: "rule.category"
-      type: string
+      tag: rename_rule_group_name
+  - rename:
+      field: crowdstrike.event.RuleDescription
+      target_field: rule.description
       ignore_missing: true
-      ignore_failure: true
-  - convert:
-      field: "crowdstrike.event.HostName"
-      target_field: "host.name"
-      type: string
+      tag: rename_rule_description
+  - rename:
+      field: crowdstrike.event.RuleFamilyID
+      target_field: rule.category
       ignore_missing: true
-      ignore_failure: true
-  - convert:
-      field: "crowdstrike.event.Ipv"
-      target_field: "network.type"
-      type: string
+      tag: rename_rule_family_id
+  - rename:
+      field: crowdstrike.event.HostName
+      target_field: host.name
       ignore_missing: true
-      ignore_failure: true
-  - convert:
-      field: "crowdstrike.event.EventType"
-      target_field: "event.code"
-      type: string
+      tag: rename_host_name
+  - rename:
+      field: crowdstrike.event.EventType
+      target_field: event.code
       ignore_missing: true
-      ignore_failure: true
+      tag: rename_event_type
   - set:
       field: network.direction
       value: ingress
+      tag: set_ingress_direction
       if: ctx?.crowdstrike?.event?.ConnectionDirection == "1"
-  - set:
-      field: source.ip
-      value: "{{crowdstrike.event.RemoteAddress}}"
-      ignore_empty_value: true
-      if: ctx?.crowdstrike?.event?.ConnectionDirection == "1"
-  - convert:
-      field: crowdstrike.event.RemotePort
-      target_field: source.port
-      type: long
+  - rename:
+      field: crowdstrike.event.RemoteAddress
+      target_field: source.ip
       ignore_missing: true
-      ignore_failure: true
-      if: ctx?.crowdstrike?.event?.ConnectionDirection == "1"
-  - set:
-      field: destination.ip
-      value: "{{crowdstrike.event.LocalAddress}}"
-      ignore_empty_value: true
-      if: ctx?.crowdstrike?.event?.ConnectionDirection == "1"
+      tag: rename_remote_address
+      if: ctx?.crowdstrike?.event?.RemoteAddress != null && ctx?.network?.direction == "ingress"
```

```suggestion
      if: ctx.crowdstrike?.event?.RemoteAddress != null && ctx.network?.direction == "ingress"
```

Some of these processors have `ctx?.`